ci, error: p2p_timeouts.py failed

Cirrus CI threw the following error in https://github.com/bitcoin-core/gui/pull/471 .
The error is not reproducible on my machine, where the same test (p2p_timeouts.py ) runs successfully.

This test is very fast, but it took 40 seconds in the CI execution.

Link: https://cirrus-ci.com/task/5600553806856192?logs=ci#L7239

```
18/232 - p2p_timeouts.py failed, Duration: 41 s
stdout:
2021-12-16T22:09:14.936000Z TestFramework (INFO): Initializing test directory /tmp/cirrus-build-177014723/ci/scratch/test_runner/test_runner_₿_🏃_20211216_220356/p2p_timeouts_212
2021-12-16T22:09:55.666000Z TestFramework.utils (ERROR): wait_until() failed. Predicate: ''''
        def test_function():
            if check_connected:
                assert self.is_connected
            return test_function_in()
'''
2021-12-16T22:09:55.666000Z TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "/tmp/cirrus-build-177014723/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/test_framework.py", line 132, in main
    self.run_test()
  File "/tmp/cirrus-build-177014723/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/p2p_timeouts.py", line 92, in run_test
    no_send_node.wait_for_disconnect(timeout=1)
  File "/tmp/cirrus-build-177014723/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/p2p.py", line 461, in wait_for_disconnect
    self.wait_until(test_function, timeout=timeout, check_connected=False)
  File "/tmp/cirrus-build-177014723/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/p2p.py", line 453, in wait_until
    wait_until_helper(test_function, timeout=timeout, lock=p2p_lock, timeout_factor=self.timeout_factor)
  File "/tmp/cirrus-build-177014723/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/util.py", line 269, in wait_until_helper
    raise AssertionError("Predicate {} not true after {} seconds".format(predicate_source, timeout))
AssertionError: Predicate ''''
        def test_function():
            if check_connected:
                assert self.is_connected
            return test_function_in()
''' not true after 40.0 seconds
```

The same has error occurred in https://github.com/bitcoin/bitcoin/pull/23807/

Link: https://cirrus-ci.com/task/6110562114338816

```
 test  2021-12-17T23:40:07.060000Z TestFramework.utils (ERROR): wait_until() failed. Predicate: '''' 
                                           def test_function():
                                               if check_connected:
                                                   assert self.is_connected
                                               return test_function_in()
                                   '''
 test  2021-12-17T23:40:07.060000Z TestFramework (ERROR): Assertion failed 
                                   Traceback (most recent call last):
                                     File "/tmp/cirrus-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/test_framework.py", line 132, in main
                                       self.run_test()
                                     File "/tmp/cirrus-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/p2p_timeouts.py", line 92, in run_test
                                       no_send_node.wait_for_disconnect(timeout=1)
                                     File "/tmp/cirrus-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/p2p.py", line 461, in wait_for_disconnect
                                       self.wait_until(test_function, timeout=timeout, check_connected=False)
                                     File "/tmp/cirrus-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/p2p.py", line 453, in wait_until
                                       wait_until_helper(test_function, timeout=timeout, lock=p2p_lock, timeout_factor=self.timeout_factor)
                                     File "/tmp/cirrus-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/util.py", line 269, in wait_until_helper
                                       raise AssertionError("Predicate {} not true after {} seconds".format(predicate_source, timeout))
                                   AssertionError: Predicate ''''
                                           def test_function():
                                               if check_connected:
                                                   assert self.is_connected
                                               return test_function_in()
                                   ''' not true after 40.0 seconds
```

--- a/test/functional/p2p_timeouts.py
+++ b/test/functional/p2p_timeouts.py
@@ -48,10 +48,13 @@ class TimeoutsTest(BitcoinTestFramework):
         self.mock_time = int(time.time())
         self.mock_forward(0)
 
-        # Setup the p2p connections
-        no_verack_node = self.nodes[0].add_p2p_connection(TestP2PConn(), wait_for_verack=False)
-        no_version_node = self.nodes[0].add_p2p_connection(TestP2PConn(), send_version=False, wait_for_verack=False)
-        no_send_node = self.nodes[0].add_p2p_connection(TestP2PConn(), send_version=False, wait_for_verack=False)
+        # Setup the p2p connections, making sure the connections are established before the mocktime is bumped
+        with self.nodes[0].assert_debug_log(['Added connection peer=0']):
+            no_verack_node = self.nodes[0].add_p2p_connection(TestP2PConn(), wait_for_verack=False)
+        with self.nodes[0].assert_debug_log(['Added connection peer=1']):
+            no_version_node = self.nodes[0].add_p2p_connection(TestP2PConn(), send_version=False, wait_for_verack=False)
+        with self.nodes[0].assert_debug_log(['Added connection peer=2']):
+            no_send_node = self.nodes[0].add_p2p_connection(TestP2PConn(), send_version=False, wait_for_verack=False)
 
         # Wait until we got the verack in response to the version. Though, don't wait for the other node to receive the
         # verack, since we never sent one

